main.py

```diff
@@ -49,257 +47,13 @@ def msgHook(ircClient: irccon.IRC, line):
-			print(f'Printed usage for !with for {user}. (Odd misses)')
-			return
-
-		misses = int(missesMatch.group(0))
-
-		hundreds = pp.getHundreds(maxCombo, misses, acc)
-
-		peppyPoints = roundString(pp.calcPP(stars, maxCombo, maxCombo - misses, hundreds, misses, acc, od, mods), 2)
-
-		modString = pp.getModString(mods)
-
-		irc.msg(user, f'{artist} - {title} [{diffName}]{modString} | {acc}%, {misses} misses: {peppyPoints}')
-		print(f'{artist} - {title} [{diffName}]{modString} | {acc}%, {misses} misses: {peppyPoints}')
-
-		conf.save(user, [lastBm, mods, acc, misses])
-
-		return
-	elif msg.find('!mods') != -1:
-		if rateLimiting.rateLimit(conf, user):
-			irc.msg(user, 'Whoa! Slow down there, bud!')
-			print(f'Rate limited user {user}.')
-			return
-
-		try:
-			mods = pp.getModVal(msg)
-
-			userBeatmap = conf.load(user)
-		except KeyError:
-			irc.msg(user, 'Please select a beatmap first! (Type /np)')
-			print(f'User {user} issued !with without a beatmap selected.')
-			return
-		except:
-			print('Usage: mods <mod1> [mod2] [mod3]...')
-			return
-
-		lastBm = userBeatmap[0]
-		acc = userBeatmap[2]
-		misses = userBeatmap[3]
-
-		artist = lastBm['artist']
-		title = lastBm['title']
-		diffName = lastBm['version']
-		creator = lastBm['creator']
-		stars = float(lastBm['difficultyrating'])
-		starsRounded = roundString(stars, 2)
-		maxCombo = int(lastBm['count_normal'])
-		od = float(lastBm['diff_overall'])
-		hp = pp.scaleHPOD(float(lastBm['diff_drain']), mods)
-		bpm = lastBm['bpm'] # Not converted to int because we only use it for printing
-		
-		hundreds = pp.getHundreds(maxCombo, misses, acc)
-
-		modString = pp.getModString(mods)
-		
-		irc.msg(user, f'{artist} - {title} [{diffName}] by {creator}, {starsRounded}* {modString} OD{od} HP{hp} BPM: {bpm} FC: {maxCombo}')
-
-		ppString = ''
-
-		# Calculate the pp for the accuracies in the tuple
-		for acc in (95.0, 96.0, 97.0, 98.0, 99.0, 100.0):
-			ppVal = roundString(pp.calcPP(stars, maxCombo, maxCombo, pp.getHundreds(maxCombo, 0, acc), 0, acc, od, mods), 2)
-			e = ' | ' # Separator
-			if acc == 95.0: # Avoid a trailing ' | '.
-				e = ''
-			ppString = f'{ppString}{e}{acc}%: {ppVal}pp'
-
-		od = pp.scaleHPOD(od, mods)
-
-		# The second line.
-		irc.msg(user, ppString)
-		print(f'OD{od} HP{hp} {starsRounded}* FC: {maxCombo}x')
-		print(ppString + '\n')
-
-		conf.save(user, [lastBm, mods, acc, misses])
-
+	if rateLimiting.rateLimit(conf, user):
+		irc.msg(user, 'Whoa! Slow down there, bud!')
+		print(f'Rate limited user {user}.')
 		return
-	elif msg.find('!discord') != -1:
-		irc.msg(user, 'The place to talk about the bot or just chat: https://discord.gg/hKXQdm2')
-		print(f'Printed discord invite for {user}.')
 
-		return
-	elif msg.find('!help') != -1:
-		irc.msg(user, 'A list of commands can be found in the [GitHub wiki](https://github.com/SarahIsWeird/taiko-bot/wiki/Commands).')
-		print(f'Printed command list for {user}.')
+	CommandHandler.Handle(user, msg, ircClient)
```

Comment:
let's just pass in `conf` and `api` right here
e.g. `CommandHandler.Handle(user, msg, ircClient, conf, api)`
then, `Handle` can pass those variables down to `commandToRun`

that way, you don't need to rebuild those variables every single time someone runs a command

Reply:
I did that for the console commands but I felt like there was a better way to do it, so I didn't for the irc commands. I guess I can put that in now and change it later when a better way is found